drivers/hmis/app/models/hmis/form/form_processor.rb

```diff
@@ -21,9 +21,9 @@ class Hmis::Form::FormProcessor < ::GrdaWarehouseBase
   belongs_to :exit, class_name: 'Hmis::Hud::Exit', optional: true, autosave: true
 
   validate :hmis_records_are_valid
-  before_save :save_enrollment
 
-  def run!
+  def run!(owner:)
+    @owner = owner
```

Comment:
This fixes a bug with validating unpersisted entry dates after processing.

Reply:
It might be cleaner to add 
```
attr_accessor :owner
```
on the model and then either set the owner before calling run:
`processor.owner = owner`  or `processor = FormProcessor.new(owner: owner); processor.run!`
or set it as you are:
`self.owner = owner`
Then you can just reference `owner` later.  
What you have is fine too, but I always wonder where the instance variable came from if it's not set in the initializer.

Reply:
The form processor already exists at this point though, I don't think we want to create a new one here.

Reply:
That's fine, just `processor.owner = owner` then or within the run method `self.owner = owner`

Reply:
I like that better, it's still a side-effect from running `run!` which isn't my favorite, but it exposes that something is going to be happening internally with `owner` which I like.  Thanks!